tests/e2e/helpers-native/Helpers.ts

```diff
@@ -6,6 +6,9 @@ export function Widget(testID: string, matcher = by.id(testID)) {
         },
         getCaption() {
             return element(by.id(`${testID}$caption`).withAncestor(matcher));
+        },
+        getText(text: string) {
+            return element(by.text(text).withAncestor(matcher));
```

Comment:
Ah, all the time spent trying to get the text and this `by.text` was available :/

Reply:
That's true. I found it later. 

But now I also realize that this should not be the first option to go for. There can be a case where the expected text is indeed not available and we're trying to find that using `by.text()`. In this scenario, the test should fail and it would eventually fail but with the reason like `element not found....` instead of mismatch while asserting the expected and actual text. Test report with more precise results is more accurate.  

So, I won't say our efforts went in vain. The only reason I don't understand why our implementation of finding using caption didn't work 🤔

Reply:
Are you saying that this isn't ideal because the following would occur:

- _does element with "hello world" exist_?
- fail, because element with "hello moon" exists_? (partial match, not exact match)

It seems fine; the exact element doesnt exist, so it should fail

Reply:
No, it would fail because the text with "hello world" was not found in the first place and not due to the mismatch.